testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -1198,36 +1198,79 @@ impl RelayerThread {
         while self.globals.keep_running() {
             let raised_initiative = self.globals.take_initiative();
             let timed_out = Instant::now() >= self.next_initiative;
-            let directive = if raised_initiative.is_some() || timed_out {
+            let mut initiative_directive = if raised_initiative.is_some() || timed_out {
                 self.next_initiative =
                     Instant::now() + Duration::from_millis(self.config.node.next_initiative_delay);
                 self.initiative()
             } else {
                 None
             };
 
-            let directive = if let Some(directive) = directive {
-                directive
-            } else {
-                match relay_rcv.recv_timeout(Duration::from_millis(
-                    self.config.node.next_initiative_delay,
-                )) {
-                    Ok(directive) => directive,
-                    Err(RecvTimeoutError::Timeout) => {
-                        continue;
```

Comment:
Maybe I'm missing something, but I'm not clear on what the substantial change in control flow is here.

My understanding is that the current implementation is roughly:

```
loop {
      let directive = if next initiative delay passed { self.initiative } else { None };
      if directive { 
        do directive
      } else { 
        relay_receive(timeout = next_initiative_delay)
      }
}
```

And the new implementation is something like:
```
loop {
      let directive = if next initiative delay passed { self.initiative } else { None };
      if directive { 
        do directive
      } else { 
        loop {
            if !received_empty_err {
               relay_try_receive()
            } else {
               relay_receive(timeout = next_initiative_delay)
            }
        }
      }
}
```

Looking at that, the only difference seems to be that the new implementation drains the relay channel before ever checking `self.initiative()`, but in practice `self.initiative()` _rarely_ returns `Some` (and it never returns `Some` if you aren't a miner). So in the existing implementation, when `self.initiative()` returns `None`, the outer loop should just keep looping to drain the receive channel. Or am I misunderstanding something in this control flow?

Reply:
No, you're right (and I'll revert this).  I'm trying to get to the bottom of why `HandleNetworkResult` directives get buffered up for longer and longer amounts of time.  I think this can be addressed better by load-shedding when the channel is full, which happens in `peer.rs` now, but I'm still testing this live.

Reply:
Reverted